pkg/agent/datamodel/helper.go

```diff
@@ -93,3 +93,19 @@ func IndentString(original string, spaces int) string {
 	}
 	return out.String()
 }
+
+func processContainerImageTag(downloadURL string) (string, error) {
+	// example URL "downloadURL": "mcr.microsoft.com/oss/kubernetes/autoscaler/addon-resizer:*",
+	// getting the data between the last / and the last :
+	parts := strings.Split(downloadURL, "/")
+	if len(parts) == 0 || len(parts[len(parts)-1]) == 0 {
+		return "", errors.New("downloadURL is not in the expected format")
+	}
+	lastPart := parts[len(parts)-1]
```

Comment:
can we actually just do `components = strings.TrimSuffix(lastPart, ":*")` instead?